### ISSUE TITLE
Convexity (full section)

### ISSUE THREAD

**Issue by SamLau95** [DS-100/textbook#13]
We need a section on convexity for gradient descent. Outline to come.

Tiff is current assigned (need one more).

This will become section 11.3, which comes right after section [11.2](https://www.textbook.ds100.org/ch11/gradient_descent_define.html).

**Outline:**

- Will gradient descent always find the theta that generates the lowest cost?
  - No; show an example of a degree 4 polynomial that gradient descent will find a local minima for
- When will gradient descent find the best theta?
  - Only when the function doesn't have local minima
  - These functions are called convex functions
- Define convex function
  - Show example of convex function and non-convex functions
  - Put convexity demo in at the end: https://calebs97.github.io/convexity_visualization/convex_demo.html
- Conclusion: when we can, we will choose a convex loss function because gradient descent will find the lowest cost instead of a local minimum.

**Comment by SamLau95** [comment]
Done! https://www.textbook.ds100.org/ch11/gradient_convexity.html
